lib/src/widgets/shape.dart

```diff
@@ -60,26 +60,31 @@ class ShapeWidget extends StatelessWidget with ChessboardGeometry {
               scale,
             ),
           ),
-        ),
-      PieceShape(
-        orig: final orig,
-        role: final role,
-        color: final color,
-        scale: final scale
-      ) =>
-        PositionedSquare(
-          size: squareSize,
-          orientation: orientation,
-          square: orig,
-          child: Image.asset(
-            'assets/piece_sets/mono/${role.uppercaseLetter}.png',
-            package: 'chessground',
-            color: color,
-            width: scale * squareSize,
-            height: scale * squareSize,
-          ),
-        ),
-    };
+        );
+      case PieceShape(
+          orig: final orig,
+          piece: final piece,
+          pieceAssets: final pieceAssets,
+          opacity: final opacity,
+          scale: final scale,
+        ):
+        {
+          final asset = pieceAssets[piece.kind]!;
+          return PositionedSquare(
+            size: size,
```

Comment:
Note: the change from `size: squareSize` to `size: size` fixes a bug introduced during earlier refactoring (`ChessboardGeometry` mixin)

Reply:
Oops! I let pass this bug? I thought I checked everything but that means we need widget tests for shape to not let it happen again.

Reply:
Yeah, I think it wasn't noticed in the app yet, because a) it's not in a official release yet, and b) we only use PieceShape for visualizing promotions in analalysis, so it's hard to miss.

I saw you already fixed in and added a test, I'll rebase my branch onto that to fix the conflict